### ISSUE TITLE
LDAP: Anonymouse bind cannot work

### ISSUE THREAD

**Issue by YasuhiroABE** [goharbor/harbor#15555]
Hello, 
I found the following issue.

**Expected behavior and actual behavior:**

For LDAP authentication, empty bind password expected to engage the anonymous bind operation. However, the default behavior of the "go-ldap v3" library does not allow the anonymous bind.

I think it causes by the difference from the old "gopkg.in/ldap.v2" library.

**Versions:**

- harbor version: 2.3.2
- kubernetes version: 1.20.7

**Appendix:**

I guess the following patch might be one of the solution.

```
diff --git a/src/pkg/ldap/ldap.go b/src/pkg/ldap/ldap.go
index 058c958d5..a9d561876 100644
--- a/src/pkg/ldap/ldap.go
+++ b/src/pkg/ldap/ldap.go
@@ -186,7 +186,11 @@ func (s *Session) SearchUser(username string) ([]model.User, error) {
 
 // Bind with specified DN and password, used in authentication
 func (s *Session) Bind(dn string, password string) error {
-       return s.ldapConn.Bind(dn, password)
+       if password == "" {
+               return s.ldapConn.UnauthenticatedBind(dn)
+       } else  {
+               return s.ldapConn.Bind(dn, password)
+       }
 }
 
 // Open - open Session, should invoke Close for each Open call
```

**Comment by stonezdj** [comment]
Yes, an empty password is allowed in LDAP but it is not allowed in Harbor for security reasons. Is there any use-case to use an empty password?

**Comment by YasuhiroABE** [comment]
Thank you for your prompt reply.
If it's intended, that's ok. 

Yes, I know some use-cases, and it is not to say these directory servers are violate against the related RFCs and information are exposed to the threat. It's depends on the network, organization structure, and policy.
But I know there are several opinions on the LDAP administration.

Let me point out one thing.
The current web ui description looks like the anonymous bind is allowed.
These two fields must be required and information should be clearly stated bind dn/pw are essential.

Thank you.
